Source/Scene/createTileMapServiceImageryProvider.js

```diff
@@ -0,0 +1,288 @@
+                rectangle.east = tilingScheme.rectangle.east;
+            }
+            if (rectangle.south < tilingScheme.rectangle.south) {
+                rectangle.south = tilingScheme.rectangle.south;
+            }
+            if (rectangle.north > tilingScheme.rectangle.north) {
+                rectangle.north = tilingScheme.rectangle.north;
+            }
+
+            // Check the number of tiles at the minimum level.  If it's more than four,
+            // try requesting the lower levels anyway, because starting at the higher minimum
+            // level will cause too many tiles to be downloaded and rendered.
+            var swTile = tilingScheme.positionToTileXY(Rectangle.southwest(rectangle), minimumLevel);
+            var neTile = tilingScheme.positionToTileXY(Rectangle.northeast(rectangle), minimumLevel);
+            var tileCount = (Math.abs(neTile.x - swTile.x) + 1) * (Math.abs(neTile.y - swTile.y) + 1);
+            if (tileCount > 4) {
+                minimumLevel = 0;
+            }
+
+            var templateUrl = joinUrls(url, '{z}/{x}/{reverseY}.' + fileExtension);
+
+            deferred.resolve({
+                url : templateUrl,
+                tilingScheme : tilingScheme,
+                rectangle : rectangle,
+                tileWidth : tileWidth,
+                tileHeight : tileHeight,
+                minimumLevel : minimumLevel,
+                maximumLevel : maximumLevel,
+                proxy : options.proxy,
+                tileDiscardPolicy : options.tileDiscardPolicy,
+                credit: options.credit
+            });
+        }
+
+        function metadataFailure(error) {
+            // Can't load XML, still allow options and defaults
+            var fileExtension = defaultValue(options.fileExtension, 'png');
+            var tileWidth = defaultValue(options.tileWidth, 256);
+            var tileHeight = defaultValue(options.tileHeight, 256);
+            var minimumLevel = defaultValue(options.minimumLevel, 0);
+            var maximumLevel = options.maximumLevel;
+            var tilingScheme = defined(options.tilingScheme) ? options.tilingScheme : new WebMercatorTilingScheme({ ellipsoid : options.ellipsoid });
+            var rectangle = defaultValue(options.rectangle, tilingScheme.rectangle);
+
+            var templateUrl = joinUrls(url, '{z}/{x}/{reverseY}.' + fileExtension);
+
+            deferred.resolve({
+                url : templateUrl,
+                tilingScheme : tilingScheme,
+                rectangle : rectangle,
+                tileWidth : tileWidth,
+                tileHeight : tileHeight,
+                minimumLevel : minimumLevel,
+                maximumLevel : maximumLevel,
+                proxy : options.proxy,
+                tileDiscardPolicy : options.tileDiscardPolicy,
+                credit: options.credit
+            });
+        }
+
+        function requestMetadata() {
+            var resourceUrl = url + 'tilemapresource.xml';
```

Comment:
@hpinkos We missed a spot during review (and apparently it's not covered by tests).  This needs to use `joinUrls` too.  Can you open a new PR with the fix and a test please. Thanks.